validation-core/src/main/scala/tagless/Solver.scala

```diff
@@ -0,0 +1,56 @@
+package jto.validation
+package v3.tagless
+
+import shapeless.{ ::, HNil, HList, Generic }
+import cats.Functor
+
+// TODO: Add helpful implicit not found message
+trait Solver[I, O] {
```

Comment:
I don't understand what this is needed for... (and reverse engineering from the types hurts my brain :D)